src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -553,6 +554,42 @@ public static MsBuildToolset GetMsBuildToolset(string userVersion, IConsole cons
             }
         }
 
+        internal static string GetNonArchitectureDirectory(string msbuildExe)
```

Comment:
The method name is bit confusing. I think `GetPlatformNeutralMsBuildPath` suits better because the method retrieves the path for the platform-neutral (non-architecture-specific) version of msbuild.exe.

Reply:
Thanks for reviewing! I think "architecture" is a more specific word, as msbuild uses `arm64` and `amd64` as the folder name, and they are processor architectures. 
While platform has a more general word, it can also stand for OS, and some other meanings.
On the other hand, since this method is not an API, we can safely change the name if more people like other naming, in the future, considering it's kind of urgent to merge this PR. What do you think?